manager/manager.go

```diff
@@ -981,7 +993,25 @@ func (m *Manager) becomeLeader(ctx context.Context) {
 	// shutdown underlying manager processes when leadership is
 	// lost.
 
-	m.allocator, err = allocator.New(s, m.config.PluginGetter)
+	// If DefaultAddrPool is null, Read from store and check if
+	// we have DefaultAddrPool info is stored in cluster object
+	if m.config.DefaultAddrPool == nil {
+		s.View(func(tx store.ReadTx) {
```

Comment:
not an expert here, but don't see any other field that requires to be re read from store directly. cc @dperny is this needed?